osquery/tables/events/linux/process_events.cpp

```diff
@@ -24,6 +24,15 @@ namespace tables {
 extern long getUptime();
 }
 
+static inline unsigned long decimalToOct(unsigned long dec){
+  unsigned long oct = 00;
+  for(auto i = 1; dec > 0; i *= 10){
```

Comment:
Consider using code below -
< for(auto i = 1; dec > 0; i *= 10){
> for(auto i = 1UL; dec > 0; i *= 10){
Reason -
'i' will we resolved as int and because of the integer overflow any value of dec >= 0x4fffffff will not be correctly converted to octal.

Besides this I would recommend using the following code instead -
 std::stringstream ss;
 ss << "0" << std::oct << dec;
r["mode"] = std::move(ss.str()); // change 'mode' column to TEXT

Two reasons -
1. Because of '0' pre-pended you can communicate to the user that its octal value(e.g. we do with hex values ''0x'). 
2. You are using integral container (unsigned long) to represent the octal values (clearly this method can not convert all the integral values to octal since octal requires more digits than decimal).

Reply:
Okay, nice. Thanks for the feedback.